torchx/components/dist.py

```diff
@@ -128,15 +129,24 @@ def ddp(
     if (script is None) == (m is None):
         raise ValueError("exactly one of --script and -m must be specified")
 
-    rep = j.split("x")
-    if len(rep) == 1:  # num replicas only
-        nnodes = 1
-        nproc_per_node = int(rep[0])
-    elif len(rep) == 2:
-        nnodes = int(rep[0])
-        nproc_per_node = int(rep[1])
+    # nnodes: number of nodes or minimum nodes for elastic launch
```

Comment:
can we pull this logic out into a helper method? that way we'll be able to test it directly

```
min_replicas, max_replicas, nproc_per_node = parse_nnodes(...)
```

Reply:
Done